### PR TITLE
Add PSR7 URI methods & make session an attribute consistently

### PR DESCRIPTION
**Pull request by markstory**
Convert the internals of URL handling in the request class to use the UriInterface implementation in diactoros. This lets us remove a bunch of code from `Request` and replace it with code from diactoros and our own ServerRequestFactory.

This set of changes looks a bit light on tests, but we have pretty extensive tests around environment parsing and those tests still pass.

Refs #9325

### CODE REVIEW

File: src/Network/Request.php [hunk 1]
```diff
@@ -348,134 +382,23 @@ protected function _processPost($data)
...
 
     /**
-     * Get the request uri. Looks in PATH_INFO first, as this is the exact value we need prepared
-     * by PHP. Following that, REQUEST_URI, PHP_SELF, HTTP_X_REWRITE_URL and argv are checked in that order.
-     * Each of these server variables have the base path, and query strings stripped off
-     *
-     * @param array $config Configuration to set.
-     * @return string URI The CakePHP request path that is being accessed.
-     */
-    protected static function _url($config)
```
Comment (lorenzo) [inline src/Network/Request.php]: I'm happy this is not here anymore

**Comment by lorenzo** [comment]
Looks good to me... thre are some CS problems, though

File: tests/TestCase/Network/RequestTest.php [hunk 2]
```diff
@@ -169,7 +169,7 @@ public function testQueryStringParsingFromInputUrl()
         $request = new Request(['url' => 'some/path?one=something&two=else']);
         $expected = ['one' => 'something', 'two' => 'else'];
         $this->assertEquals($expected, $request->query);
-        $this->assertEquals('some/path?one=something&two=else', $request->url);
+        $this->assertEquals('some/path', $request->url);
```
Comment (saeideng) [inline tests/TestCase/Network/RequestTest.php]: before this ,`$request->url` returned `some/path` not `some/path?one=something&two=else`
but there is this  test and test tools not able find that?why?

File: src/Network/Request.php (hunk 1 above)
Comment (markstory) [inline src/Network/Request.php]: Me too. I should have been smarter in 2.x days and created a RequestFactory.

File: tests/TestCase/Network/RequestTest.php (hunk 2 above)
Comment (markstory) [inline tests/TestCase/Network/RequestTest.php]: I feel that the old code was 'wrong' before. The 'url' (which is what Router sees) should not include query string arguments. It should be the path segment only. The name `url` is of course not fitting and will be going away :smile:

**Comment by markstory** [comment]
PHPCS errors should be good now.

**Comment by lorenzo** [comment]
good job!

**Pull request merged**